StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1648InheritDocMustBeUsedWithInheritingClass.cs

```diff
@@ -159,6 +159,56 @@ private static void HandleMemberDeclaration(SyntaxNodeAnalysisContext context)
                 }
             }
 
+            if (memberSyntax is ConstructorDeclarationSyntax constructorDeclarationSyntax)
+            {
+                ISymbol symbol = context.SemanticModel.GetDeclaredSymbol(constructorDeclarationSyntax, context.CancellationToken);
+
+                if (symbol is IMethodSymbol constructorMethodSymbol && constructorMethodSymbol.ContainingType is INamedTypeSymbol enclosingNamedTypeSymbol)
```

Comment:
There is too much code here for my taste. Move most of this to a (or multiple) helper method to make it easier to read. Possibly in NamedTypeExtensions, which already has similar functionality, but since this class would be the only caller right now, putting it here might be preferable.

Candidates for extraction:
- ConstructorSignatureMatchesBaseType(IMethodSymbol)
- bool HasMatchingSignatures(IMethodSymbol, IMethodSymbol)